x-pack/plugins/lens/public/indexpattern_datasource/loader.test.ts

```diff
@@ -287,10 +300,51 @@ describe('loader', () => {
       });
     });
 
+    it('should load a default state when lastUsedIndexPatternId is not found in indexPatternRefs', async () => {
```

Comment:
Will it also write back to localStorage overwriting the missing index pattern?

Reply:
Good idea! I'll add it.

Reply:
Actually, that made me think
1. (scenario you mentioned) lastUsedIndexPatternId is not found in kibana - we write back to localStorage overwriting the missing index pattern
2. (second scenario) we open saved visualization - should we also write back to localStorage? I think so 🤔  -for now we only do it on index pattern change